samples/cellular/modem_shell/README.rst

```diff
@@ -464,80 +464,14 @@ Modem traces
 
 MoSh command: ``modem_trace``
 
+Enable the ``modem_trace`` command using the :kconfig:option:`CONFIG_NRF_MODEM_LIB_SHELL_TRACE` Kconfig option.
```

Comment:
Earlier, the `modem_trace` command has been available in modem_shell when `CONFIG_NRF_MODEM_LIB_TRACE` is enabled. Now it needs to be enabled separately using this Kconfig option. There's a possibility to change the default value for `CONFIG_NRF_MODEM_LIB_SHELL_TRACE` in modem_shell Kconfig file to retain the old behavior. @trantanen?

Reply:
This is because of a request from @lemrey in this PR.

Reply:
Yes, I know. But we could still change the default value in modem_shell.

Reply:
I'm really 50-50 on this and don't have a strong opinion. I'd expect that more common is to use the snippet for UART tracing and shell commands for modem tracing could be behind few Kconfig options for those who want to use them.

The documentation should mention that both `CONFIG_NRF_MODEM_LIB_TRACE` and `CONFIG_NRF_MODEM_LIB_SHELL_TRACE` must be enabled for `modem_trace` command to be available.

Reply:
> Yes, I know. But we could still change the default value in modem_shell.

That's true. I forgot about that. Should I re-define the default for modem_shell @tokangas @trantanen ?

Reply:
I discussed with @tokangas. Let's NOT re-define the default values for modem shell.
However, update the description so that it mentions what else is needed with `CONFIG_NRF_MODEM_LIB_SHELL_TRACE` to get it enabled. I don't know if it's better to talk about the snippet or just `CONFIG_NRF_MODEM_LIB_TRACE`.

Reply:
> However, update the description so that it mentions what else is needed with CONFIG_NRF_MODEM_LIB_SHELL_TRACE to get it enabled. I don't know if it's better to talk about the snippet or just CONFIG_NRF_MODEM_LIB_TRACE.

Is it common to mention kconfig dependencies in the readme? It's already listed when clicking the kconfig name and in the «See [Modem trace module](https://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-13725/nrf/libraries/modem/nrf_modem_lib/nrf_modem_lib_trace.html#modem-trace-module) for more information on how to configure modem tracing...» link just below.

Reply:
If the user has to explicitly enable the CONFIG_NRF_MODEM_LIB_TRACE along with the CONFIG_NRF_MODEM_LIB_SHELL_TRACE Kconfig option to enable the ``modem_trace`` command, then I think this is good to be documented.

Reply:
Maybe it's not needed. I was just thinking that because the missing `Kconfig` may go unnoticed as a warning during the build (there is no failure), it might be worth mentioning.

`modem_trace` command is special in many ways from modem shell perspective because it's the only command included in the build configuration, documented and not implemented in modem shell code.

Having said that, I'll leave it to you to decide.

Reply:
@divipillai It depends on how they're using it. If they're using one of the snippets for modem tracing it's already enabled in the snippet. Otherwise they'll need to specify it manually. I added it to the docs now.